cloudflare/resource_cloudflare_ip_list.go

```diff
@@ -0,0 +1,170 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/pkg/errors"
+	"regexp"
+)
+
+func resourceCloudflareIPList() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareIPListCreate,
+		Read:   resourceCloudflareIPListRead,
+		Update: resourceCloudflareIPListUpdate,
+		Delete: resourceCloudflareIPListDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareIPListImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile("^[0-9a-z_]+$"), "IP List name must only contain lowercase letters, numbers and underscores"),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"kind": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{"ip"}, false),
+				Required:     true,
+			},
+			"item": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     listItemElem,
+			},
+		},
+	}
+}
+
+var listItemElem = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"id": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"value": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"comment": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+	},
+}
+
+func resourceCloudflareIPListCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	list, err := client.CreateIPList(context.Background(), d.Get("name").(string), d.Get("description").(string), d.Get("kind").(string))
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("error creating IP List %s", d.Get("name").(string)))
+	}
+
+	d.SetId(list.ID)
+
+	if items, ok := d.GetOk("item"); ok {
+		IPListItems := buildIPListItemsCreateRequest(items.(*schema.Set).List())
+		_, err = client.CreateIPListItems(context.Background(), d.Id(), IPListItems)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("error creating IP List Items"))
+		}
+	}
+
+	return resourceCloudflareIPListRead(d, meta)
+}
+
+func resourceCloudflareIPListImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	listID := d.Id()
```

Comment:
I suspect we will need the ID to be a composite ID of `accountID/listID` to allow people with access to multiple accounts to determine which one they would like to manage.

Reply:
Oh thanks for raising this. I was a victim to `DefaultFunc: schema.EnvDefaultFunc("CLOUDFLARE_ACCOUNT_ID", nil),` in provider.go ... I'll get the whole PR ready by - hopefully - tomorrow!

Reply:
No rush! I just finally had some time this morning to review some PRs and yours was at the top of the list 🙂

Reply:
This is fixed